src/main/java/busynessmanager/BusynessManager.java

```diff
@@ -0,0 +1,398 @@
+import static busynessmanager.constants.Constants.BM_EXIT_KEYWORD;
+import static busynessmanager.constants.Constants.BM_EXIT_MESSAGE;
+import static busynessmanager.constants.Constants.BM_ID_ASSERTION_FAIL_MESSAGE;
+import static busynessmanager.constants.Constants.BM_PASSWORD_NULL_ASSERTION_FAIL_MESSAGE;
+import static busynessmanager.constants.Constants.BM_NO_CREDENTIALS_MESSAGE;
+import static busynessmanager.constants.Constants.BM_NO_DATA_MESSAGE;
+import static busynessmanager.constants.Constants.BM_SAVE_SUCCESS_MESSAGE;
+import static busynessmanager.constants.Constants.BM_SAVE_FAIL_MESSAGE;
+import static busynessmanager.constants.Constants.BM_LOAD_SUCCESS_MESSAGE;
+import static busynessmanager.constants.Constants.BM_LOAD_FAIL_MESSAGE;
+
+
+/**
+ * The main class for the Busyness Manager application.
+ * Handles user authentication, business setup, and command execution.
+ */
+public class BusynessManager {
+    private static final String DATA_FOLDER = "data";
+    private static final String BUSINESS_INFO_FILE = DATA_FOLDER + "/BusinessInfo.txt";
+
+    public enum BusinessType {
+        FNB, RETAIL
+    }
+
+    private Credentials credentials;
+    private final InventoryManager inventoryManager;
+    private final CommandParser commandParser;
+
+
+    /**
+     * Constructs a BusynessManager instance and initializes the necessary managers.
+     * Also loads business credentials if available.
+     */
+    public BusynessManager() {
+        inventoryManager = new InventoryManager();
+        SalesManager salesManager = new SalesManager(inventoryManager);
+        RevenueCalculator revenueCalculator = new RevenueCalculator(salesManager);
+        SearchManager searchManager = new SearchManager(inventoryManager);
+        commandParser = new CommandParser(inventoryManager, salesManager, revenueCalculator, searchManager);
+        loadBusinessData();
+    }
+
+
+    /**
+     * The entry point of the program.
+     *
+     * @param args Command-line arguments (not used).
+     */
+    public static void main(String[] args) {
+        BusynessManager manager = new BusynessManager();
+        manager.start();
+    }
+
+    /**
+     * Starts the application by handling user authentication or first-time setup.
+     */
+    private void start() {
+        Scanner scanner = new Scanner(System.in);
+        UI.printMessage(BM_WELCOME_MESSAGE);
+
+        if (credentials == null) {
+            UI.printMessageWithoutNewline(BM_FIRST_SETUP_CHECK_MESSAGE);
+            String response = scanner.nextLine();
+
+            if (response.equalsIgnoreCase(BM_FIRST_SETUP_APPROVAL)) {
+                firstTimeSetup(scanner);
+            } else {
+                UI.printMessage(BM_EXIT_MESSAGE);
+                return;
+            }
+        } else {
+            login(scanner);
+        }
+
+        run(scanner);
+    }
+
+    /**
+     * Handles user login.
+     *
+     * @param scanner The Scanner object for user input.
+     */
```

Comment:
May I please check why do you have user login if the application is a single user application?
PS- important to note that your application is supposed to be single-user only (It is one of the tp constraints)